### PR TITLE
Ui/TRA-4586_add-insertion-filter-to-settings-page

### PR DESCRIPTION
**Pull request by leon-up9**

### CODE REVIEW

File: ui-common/src/components/TrafficViewer/Filters.tsx [hunk 1]
```diff
@@ -47,18 +54,59 @@ export const modalStyle = {
...
 
-export const QueryForm: React.FC<QueryFormProps> = ({backgroundColor, reopenConnection}) => {
+export const CodeEditorWrap: FC<QueryFormProps> = ({ query, onQueryChange, validateQuery, onValidationChanged }) => {
+    const [queryBackgroundColor, setQueryBackgroundColor] = useState("#f5f5f5");
+    const handleQueryChange = useMemo(
+        () =>
+            debounce(async (query: string) => {
+                if (!query) {
+                    setQueryBackgroundColor("#f5f5f5");
+                    onValidationChanged && onValidationChanged({ query: query, message: "", valid: true })
```
Comment (AmitUp9) [inline ui-common/src/components/TrafficViewer/Filters.tsx:64]: why onCalidationChanged Twice?

Comment (leon-up9) [reply ui-common/src/components/TrafficViewer/Filters.tsx:64]: you mean called twice?
what would you suggest?

Comment (AmitUp9) [reply ui-common/src/components/TrafficViewer/Filters.tsx:64]: I missed the "?" in interface. I got it now. Is there a case where you don't want to use "onValidationChanged" in CodeEditorWrap?

File: ui-common/src/components/TrafficViewer/Filters.tsx [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
         <QueryForm
-            backgroundColor={backgroundColor}
+            query={query}
             reopenConnection={reopenConnection}
-        />
+            onQueryChange={(query) => { setQuery(query?.trim()); }} validateQuery={api?.validateQuery} />
     </div>;
 };
 
+type OnQueryChnage = { valid: boolean, message: string, query: string }
```
Comment (lirazyehezkel) [inline ui-common/src/components/TrafficViewer/Filters.tsx:33]: OnQueryChange

File: ui-common/src/components/TrafficViewer/Filters.tsx [hunk 3]
```diff
@@ -47,18 +54,59 @@ export const modalStyle = {
...
 
     const [openModal, setOpenModal] = useState(false);
 
     const handleOpenModal = () => setOpenModal(true);
     const handleCloseModal = () => setOpenModal(false);
 
-    const handleChange = async (e) => {
-        setQuery(e.target.value.trim());
+    const handleChange = async (queryObj) => {
+        onQueryChange(queryObj);
```
Comment (lirazyehezkel) [inline ui-common/src/components/TrafficViewer/Filters.tsx:109]: I think this is a redundant func, you can call onQueryChange directly

**Pull request merged**